singer/transform.py

```diff
@@ -271,7 +274,23 @@ def _transform(self, data, typ, schema, path):
                 return False, None
 
             return True, data
+        elif schema.get("format") == "singer.decimal":
+            if data is None:
+                return False, None
 
+            if isinstance(data, (str, float, int)):
+                try:
+                    return True, str(decimal.Decimal(str(data)).normalize())
+                except:
+                    return False, None
+            elif isinstance(data, decimal.Decimal):
+                # NB: This treats all NaN values as "null"
+                if data.is_nan() or data.is_snan() or data.is_qnan():
```

Comment:
This case should be removed (leaving just the "else" case) because we don't want to return `True, None` for something that is not the null schema (above).

Reply:
When we pass an 'sNaN' through, it can be parsed as a decimal but can't be normalized.

Should we still consider the case where `data.is_snan()' and just not normalize it if so?

Reply:
So when we pass a decimal.Decimal('NaN') through, it gets picked up as a string and 'NaN' is returned.
But when we pass a decimal.Decimal('sNaN') through, it fails as a string and gets picked up as a null and 'None' is returned.

Do we want all 'NaN's to be returned as 'NaN's, or Nones? If we want them to be returned as Nones, we'll need the case above.

Reply:
Hm, yeah. I'm not quite sure exactly how this should be handled. The big issue is that there is no BigDecimal representation for NaN and +/-Infinity in Java, so this will likely just be rejected, unless it's handled in the parsing downstream.

It might be best just to pass these values along and let them be handled downstream. The behavior that just needs to be agreed upon is what should the input/output be in the cases of `{"type": "string", "format": "singer.decimal"}` and `{"type":  ["null","string"], "format": "singer.decimal"}`.

My gut says we should just pass them along as strings until we have some real idea of how someone would interpret those values, then parse them accordingly downstream.

Reply:
Just removing this case would probably be for the best, but keeping the test that confirms `Decimal('NaN')` and others are converted to their string counterparts.

If it turns out to be an issue (with real data examples), we can talk about handling these cases.

Reply:
cool, so with no special cases, NaN will work but sNaNs wont
but according to the docs you sent us, java doesn’t know how to interpret sNaNs anyways, so I think that means we don’t have to worry about it